classes/note/Note.php

```diff
@@ -3,176 +3,148 @@
     {
-        return $this->getData('contents');
+        return Attribute::make(
+            get: function () {
+                return Repo::user()->get($this->userId, true);
+            },
+        );
     }
 
     /**
-     * set note contents
+     * Compatibility function for including note IDs in grids.
      *
-     * @param string $contents
+     * @deprecated 3.5 Use $model->id instead. Can be removed once the DataObject pattern is removed.
      */
-    public function setContents($contents)
+    public function getId(): int
     {
-        $this->setData('contents', $contents);
+        return $this->id;
     }
 
     /**
-     * get note title
+     * Compatibility function for including notes in grids.
      *
-     * @return string
+     * @deprecated 3.5. Use $model or $model->$field instead. Can be removed once the DataObject pattern is removed.
      */
-    public function getTitle()
+    public function getData(?string $field)
     {
-        return $this->getData('title');
+        return $field ? $this->$field : $this;
     }
 
+    // Scopes
+
     /**
-     * set note title
-     *
-     * @param string $title
+     * Scope a query to only include notes with a specific user ID.
      */
-    public function setTitle($title)
+    public function scopeWithUserId(Builder $query, int $userId): Builder
     {
-        $this->setData('title', $title);
+        return $query->where('userId', $userId);
     }
 
     /**
-     * get note type
-     *
-     * @return int
+     * Scope a query to only include notes with a specific assoc type and assoc ID.
      */
-    public function getAssocType()
+    public function scopeWithAssoc(Builder $query, int $assocType, int $assocId): Builder
     {
-        return $this->getData('assocType');
+        return $query->where('assoc_type', $assocType)
+                     ->where('assoc_id', $assocId);
     }
 
     /**
-     * set note type
-     *
-     * @param int $assocType
+     * Scope a query to only include notes with a specific type.
      */
-    public function setAssocType($assocType)
+    public function scopeWithType(Builder $query, int $type): Builder
     {
-        $this->setData('assocType', $assocType);
+        return $query->where('type', $type);
     }
 
     /**
-     * get note assoc id
-     *
-     * @return int
+     * Scope a query to only include notes with a specific type.
      */
-    public function getAssocId()
+    public function scopeWithContents(Builder $query, string $contents): Builder
     {
-        return $this->getData('assocId');
+        return $query->where('contents', $contents);
     }
 
     /**
-     * set note assoc id
-     *
-     * @param int $assocId
+     * Scope a query to a specific sort order.
      */
-    public function setAssocId($assocId)
+    public function scopeWithSort(Builder $query, int $orderBy = NOTE_ORDER_DATE_CREATED, int $sortDirection = DAO::SORT_DIRECTION_DESC): Builder
```

Comment:
```suggestion
    public function scopeWithSort(Builder $query, int $orderBy = static::NOTE_ORDER_DATE_CREATED, int $sortDirection = DAO::SORT_DIRECTION_DESC): Builder
```

Reply:
Setting this to static results in `static::" is not allowed in compile-time constants` errors so I've left this one out.

Reply:
Sorry, you're right that `static::` won't work -- but shouldn't it be `self::`? It's now a class constant, and I think you removed the old backwards compatibility defines for these constants, so just referring to `NOTE_ORDER_DATE_CREATED` without qualification should result in an `Uncaught Error: Undefined constant` error.

Reply:
Yes, I forgot to note that I've set it to `self::` instead